From 588c8f434d33a338a20d83f0418533b2d9faf79d Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Sat, 24 Mar 2018 22:44:10 -0700
Subject: [PATCH 2/4] c-ada-spec: Workaround for impurity detection

Our scripts for detecting if bad directories paths have been written
into our gcc binaries by accident will always get tripped up over these.
We know this use of the directory is fine so we are effectively
whitelisting it.
---
 gcc/c-family/c-ada-spec.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gcc/c-family/c-ada-spec.c b/gcc/c-family/c-ada-spec.c
index c85e6060d3e..477d0ae272a 100644
--- a/gcc/c-family/c-ada-spec.c
+++ b/gcc/c-family/c-ada-spec.c
@@ -938,8 +938,8 @@ get_ada_package (const char *file)
   int i;
   size_t plen;
 
-  s = strstr (file, "/include/");
-  if (s)
+  s = strstr (file, "/includ");
+  if (s && s[7] == 'e' && s[8] == '/')
     base = s + 9;
   else
     base = lbasename (file);
-- 
2.21.0.1020.gf2820cf01a-goog

